### ISSUE TITLE
Question: Chapter PicInfo items are not stored to file when using `m4b`?

### ISSUE THREAD

**Issue by sandreas** [Zeugma440/atldotnet#145]
## The problem

Chapter pictures and descriptions are not stored in `m4b` files.

## Environment

* 4.3.0 (4.5.0 does not work, see issue #144)
* macOS / Linux

## Code To Reproduce Issue [ Good To Have ]
```c#
var t = new Track("a-file.m4b");
Console.WriteLine(t.Chapters.Count);
if(t.Chapters.Count > 0)    {
    if(t.Chapters.First().Picture == null) {
        Console.WriteLine("no picture for first chapter");
    }
    var picInfo = PictureInfo.fromBinaryData(await File.ReadAllBytesAsync("a-file.chapters[1].jpg"));
    picInfo.PicType = PictureInfo.PIC_TYPE.Generic;
    picInfo.Description = "Chapter cover";
    picInfo.ComputePicHash();
    t.Chapters.First().Picture = picInfo;
    if(t.Save()) {
        Console.WriteLine("file saved");
    }
}
```

The sample shows:
```
17
no picture for first chapter
file saved
```
no matter how often I run it... after the first run it should tell me, that there is a picture for the first chapter, shouldn't it?

**Comment by Zeugma440** [comment]
Chapter pictures are implemented for ID3v2 only.

I haven't found out if they are supported by other formats.

I'd be delighted to have a sample M4B/MP4/M4A that has working chapter pictures, if you have any !

**Comment by sandreas** [comment]
> I haven't found out if they are supported by other formats.

Ok, I'll tell you, what I found out so far:
- it seems to be necessary to tell the mp4 tracks/container that this is a menu track for the main track
- Maybe this is an Apple specific feature (https://en.wikipedia.org/wiki/Enhanced_podcast)
- More details: https://github.com/sandreas/m4b-tool/issues/21

> I'd be delighted to have a sample M4B/MP4/M4A that has working chapter pictures, if you have any !

```bash
wget http://media.baseballphd.net/BPHD-Wichita-Episode-431.m4a
```

This file has 2 Chapters:
- Introduction
- Steve Schuster

If I use VLC Media Player and select the second chapter (Playback, Chapter, Steve Schuster), it shows another image. I did not find out how, but in VLC it works.

**Comment by Zeugma440** [comment]
Awesome. Thanks a lot.

I'm first gonna ship a release with this week's bugfixes, and start analyzing that afterwards.

**Comment by Zeugma440** [comment]
There are 4 track on your sample :
- Track 1 : Text (Chapters with name only; 2 entries)
- Track 2 : Text (Chapters with name and URL; 3 entries; the latter is real tiny, has no title and lasts a few milliseconds)
- Track 3 : Audio
- Track 4 : Images (JPEG data; 3 entries -> podcast logo, photograph of Steve Schuster and podcast logo again)

I wonder what track 2 is used for. Do you know if it is mandatory to display pictures from track 4 ?

**Comment by sandreas** [comment]
> Do you know if it is mandatory to display pictures from track 4 ?

Unfortunately no, I have no idea what this is about. I've never seen such a file nor I have an idea, what part of the spec is used to store this.

Even VLC is not too happy with the file, only when I switch chapters the cover changes - if I jump to a track position AFTER the chapter switch, the cover does not change... very strange. I'll try to find more examples for chapters with covers, as soon as I find more, I'll link them here in the issue.

**Comment by Zeugma440** [comment]
Finally nailed it !

There's no need for a 2nd text track at all. I was using a wrong atom all along to describe data inside the picture track. Now it works on VLC and Windows Media Player (same behaviour as your track and the other I found - picture gets loaded when you manually select the chapter).

Looks like the whole thing behaves a lot better when you use pictures with the same dimensions for each chapter.

**Comment by Zeugma440** [comment]
Published in today's v4.07
